src/RobotInterface/include/BipedalLocomotion/RobotInterface/ISensorBridge.h

```diff
@@ -0,0 +1,296 @@
+{
+    std::vector<std::string> jointsList; /**< list of joints attached to the bridge */
+    std::vector<std::string> IMUsList;   /**< list of IMUs attached to the bridge */
+    std::vector<std::string> linearAccelerometersList; /**< list of linear accelerometers attached to the bridge */
+    std::vector<std::string> gyroscopesList; /**< list of gyroscopes attached to the bridge */
+    std::vector<std::string> orientationSensorsList; /**< list of orientation sensors attached to the bridge */
+    std::vector<std::string> magnetometersList; /**< list of magnetometers attached to the bridge */
+    std::vector<std::string> sixAxisForceTorqueSensorsList; /**< list of six axis force torque sensors attached to the bridge */
+    std::vector<std::string> threeAxisForceTorqueSensorsList; /**< list of three axis force torque sensors attached to the bridge */
+    std::vector<std::string> cartesianWrenchesList; /**< list of cartesian wrench streams attached to the bridge */
+    std::vector<std::string> rgbCamerasList; /**< list of rgb cameras attached to the bridge */
+    std::vector<std::string> depthCamerasList; /**< list of depth cameras attached to the bridge */
+};
+
+/**
+ * Sensors bridge interface.
+ */
+class ISensorBridge
+{
+public:
+    using unique_ptr = std::unique_ptr<ISensorBridge>;
+
+    using shared_ptr = std::shared_ptr<ISensorBridge>;
+
+    using weak_ptr = std::weak_ptr<ISensorBridge>;
+
+    /**
+     * Initialize estimator
+     */
+    virtual bool initialize(std::weak_ptr<BipedalLocomotion::ParametersHandler::IParametersHandler> handler) = 0;
+
+    /**
+     * Get joints list
+     */
+    virtual const std::vector<std::string> getJointsList() = 0;
+
+    /**
+     * Get imu sensors list
+     */
+    virtual const std::vector<std::string> getIMUsList() = 0;
+
+    /**
+     * Get linear accelerometers list
+     */
+    virtual const std::vector<std::string> getLinearAccelerometersList() = 0;
+
+    /**
+     * Get gyroscopes list
+     */
+    virtual const std::vector<std::string> getGyroscopesList() = 0;
+
+    /**
+     * Get orientation sensors list
+     */
+    virtual const std::vector<std::string> getOrientationSensorsList() = 0;
+
+    /**
+     * Get magnetometers sensors list
+     */
+    virtual const std::vector<std::string> getMagnetometersList() = 0;
+
+    /**
+     * Get 6 axis FT sensors list
+     */
+    virtual const std::vector<std::string> getSixAxisForceTorqueSensorsList() = 0;
+
+    /**
+     * Get 6 axis FT sensors list
+     */
+    virtual const std::vector<std::string> getThreeAxisForceTorqueSensorsList() = 0;
+
+   /**
+     * Get cartesian wrenches list
+     */
+    virtual const std::vector<std::string> getCartesianWrenchesList() = 0;
+
+    /**
+     * Get rgb cameras list
+     */
+    virtual const std::vector<std::string> getRGBCamerasList() = 0;
+
+    /**
+     * Get depth cameras list
+     */
+    virtual const std::vector<std::string> getDepthCamerasList() = 0;
```

Comment:
Oh yes.. that was the initial intention, but I changed my mind since it would depend on whether the implementation will hold the data structure in their class. So I thought return by value would be better, but I forgot to remove the const.

Reply:
This would be a big waste of memory. I think that, in this case, the possible choices would be:
- return by reference
- have a reference to the vector as an input to the method

Reply:
I think we can go for the latter without having to impose the maintenance of those data structures by the implementation class.

Reply:
As you prefer!

Reply:
But actually, to be extra safe, just not return the reference of a string that "might" be local, I will use the second choice.

Reply:
> But actually, to be extra safe, just not return the reference of a string that "might" be local, I will use the second choice.

I guess you meant ``vector`` and not ``string``, but yeah, I agree there is the risk that in a faulty implementation a reference to a local object is returned.

Reply:
Actually, I realized later that these methods are probably called just once, so even return them by copy is not a big deal. So, feel free to return directly the vector if you believe the interface would be cleaner.

Reply:
Actually I already changed them to be passed by reference and then copied. We can go with this choice, which can also handle failure cases through the boolr eturn.

Reply:
Sounds good!

Reply:
Done in https://github.com/dic-iit/bipedal-locomotion-framework/pull/87/commits/666670f08e102f679fff287bc896cadc7b91de14